parser/parser.go

```diff
@@ -581,6 +598,18 @@ func (p *parser) validateApp() {
 	}
 }
 
+func (p *parser) validateTypeDoesntUseConfigTypes(pos token.Pos, param *est.Param) {
+	err := schema.Walk(p.decls, param.Type, func(n any) error {
+		if _, ok := n.(*schema.ConfigValue); ok {
+			return errors.New("config.Value")
+		}
+		return nil
+	})
+	if err != nil {
+		p.errf(pos, "type %s can only be used in data types used by config.Load and cannot be used for API response/response parameters or PubSub message types", err.Error())
```

Comment:
Maybe accept `API response/response parameters or PubSub message` as a `context string` argument so we can be more specific in the error and not have to list all possible cases where it's not allowed